File: javascript/src/core/exceptions.js
export class FlyqlError extends Error {
    constructor(message) {
        super(message)
        this.name = 'FlyqlError'
    }
}

export class ParserError extends FlyqlError {
    constructor(message, errno, range = null) {
        super(message)
        this.errno = errno
        this.range = range
    }

    toString() {
        return this.message
    }

    toRepresentation() {
        return this.toString()
    }
}

export class KeyParseError extends FlyqlError {
    constructor(message, range) {
        super(message)
        this.range = range
    }
}

File: python/flyql/core/exceptions.py
from typing import TYPE_CHECKING, Optional

if TYPE_CHECKING:
    from flyql.core.range import Range


class FlyqlError(Exception):
    def __init__(self, message: str) -> None:
        self.message = message
        super().__init__(message)


class KeyParseError(FlyqlError):
    def __init__(self, message: str, range: "Range") -> None:
        super().__init__(message)
        self.range = range
